src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -80,11 +81,37 @@ public CursorsService(final ZooKeeperHolder zkHolder,
 
         validateStreamId(cursors, streamId, subscriptionId);
 
+        LOG.debug("[COMMIT_CURSORS] stream IDs validation finished");
+
+        final Iterator<EventTypePartition> distinctPartitions = cursors.stream()
+                .map(c -> new EventTypePartition(c.getEventType(), c.getPartition()))
+                .distinct()
+                .iterator();
+
+        final HashMap<EventTypePartition, Iterator<Boolean>> partitionCommits = new HashMap<>();
+        while (distinctPartitions.hasNext()) {
```

Comment:
Usually, iterators are used for changing iterating collection, while you iterate it. Probably it can be replaced with foreach loop.